### PR TITLE
Allow disabling parallel execution for jobs

### PR DESCRIPTION
**Pull request by afnanenayet**
Add an option so jobs (whether scheduled or not) can be disallowed from running concurrently

### CODE REVIEW

File: job.go [hunk 1]
```diff
@@ -22,17 +22,20 @@ func IsJobCancelled(ctx context.Context) bool {
 	}
 }
 
-// EventTriggerListenersProvider is a type that enables or disables logger listeners.
+// EventTriggerListenersProvider is a type that enables or disables logger
+// listeners.
```
Comment (mat285) [inline job.go]: this is fine on one line

File: job.go [hunk 2]
```diff
@@ -22,17 +22,20 @@ func IsJobCancelled(ctx context.Context) bool {
...
 type EventTriggerListenersProvider interface {
 	ShouldTriggerListeners() bool
 }
 
-// EventShouldWriteOutputProvider is a type that enables or disables logger output for events.
+// EventShouldWriteOutputProvider is a type that enables or disables logger
```
Comment (mat285) [inline job.go]: same ^

File: job.go [hunk 3]
```diff
@@ -22,17 +22,20 @@ func IsJobCancelled(ctx context.Context) bool {
...
+// output for events.
 type EventShouldWriteOutputProvider interface {
 	ShouldWriteOutput() bool
 }
 
-// EnabledProvider is an optional interface that will allow jobs to control if they're enabled.
+// EnabledProvider is an optional interface that will allow jobs to control if
+// they're enabled.
```
Comment (mat285) [inline job.go]: and this

File: job_manager.go [hunk 4]
```diff
@@ -47,6 +47,7 @@ func New() *JobManager {
 		nextRunTimes:          map[string]*time.Time{},
 		disabledJobs:          collections.SetOfString{},
 		enabledProviders:      map[string]func() bool{},
+		disabledParallelJobs:  collections.SetOfString{},
```
Comment (wcharczuk) [inline job_manager.go]: for now i would start with an interface, `RunParallelProvider` that is a method that job types can implement. i don't imagine we'll need job types running in parallel in one place but not in others, and it's easier to push that configuration out on to the job creator

**Comment by afnanenayet** [comment]
@mat285 it just cancels the run and it'll run again at the next scheduled run

**Pull request merged**